src/block/Jukebox.php

```diff
@@ -89,10 +89,8 @@ public function stopSound() : void{
 		$this->getPos()->getWorld()->addSound($this->getPos(), new RecordStopSound());
 	}
 
-	public function onBreak(Item $item, ?Player $player = null): bool
-	{
-		$this->ejectRecord();
-		return parent::onBreak($item, $player);
+	public function getDrops(Item $item): array{
+		return $this->record !== null ? [$this->record] : [];
```

Comment:
This breaks the regular jukebox block drop.

Reply:
You also don't want to return the original item, since the caller might modify it.

Reply:
I will get there 😊